src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -55,7 +56,7 @@ public static class EnumerableMappingBuilder
 
         // try linq mapping: x.Select(Map).ToArray/ToList
         // if that doesn't work do a foreach with add calls
-        var (canMapWithLinq, collectMethodName) = ResolveCollectMethodName(ctx, elementMapping.IsSynthetic);
+        var (canMapWithLinq, collectMethodName) = ResolveCollectMethodName(ctx);
```

Comment:
Not directly related to this MR: Shouldn't we try to use LINQ as a last resort and try to use foreach first?

Reply:
There is an issue for this: https://github.com/riok/mapperly/issues/433